src/types/index.ts

```diff
@@ -26,3 +28,39 @@ export interface KafkaControlPlaneResponse {
   title: string;
   body: string;
 }
+
+export interface SlackMessage {
+  type: 'slack';
+  channels: string[];
+  text: string;
+  blocks?: KnownBlock[] | Block[];
+}
+
+export interface DatadogEvent {
+  type: 'datadog';
+  title: string;
+  text: string;
+  tags: string[];
+  alertType: EventAlertType;
+}
+
+export interface JiraEvent {
+  type: 'jira';
+  projectId: string;
+  title: string;
+}
+
+export type GenericEvent = {
+  source: string;
+  timestamp: number;
+  data: (DatadogEvent | JiraEvent | SlackMessage | ServiceSlackMessage)[];
+};
+
+// Currently only used for Slack notifications since
+// service registry only contains Slack channels (and not DD or Jira or others)
+export interface ServiceSlackMessage {
+  type: 'service_notification';
+  service_name: string; // Official service registry service id
+  text: string;
+  blocks?: KnownBlock[] | Block[];
+}
```

Comment:
This seems the same as the slack message, except that you can provide the channel via a service name.
Doing it this way would mean we would duplicate all the methods that can take a destination from the service registry.

Why not abstracting the concept of SlackChannel, and pass it here https://github.com/getsentry/eng-pipes/pull/879/files#diff-a9ba9cbedd19c9f66d564d2e89912890209b98f0a7ef19187877d2587300e476R34 .

You can create two implementations: SlackChannel and ServiceSlackChannel. One contains the channel name and the other contains the service name.